src/storage/src/store.rs

```diff
@@ -114,47 +130,69 @@ pub trait StateStore: Send + Sync + 'static + Clone {
     /// - A version of a kv pair. kv pair associated with larger `Epoch` is guaranteed to be newer
     ///   then kv pair with smaller `Epoch`. Currently this version is only used to derive the
     ///   per-key modification history (e.g. in compaction), not across different keys.
+    ///
+    /// `table_id` is used to specify the storage table to write to.
     fn ingest_batch(
         &self,
         kv_pairs: Vec<(Bytes, StorageValue)>,
-        epoch: u64,
+        epoch: HummockEpoch,
+        table_id: StorageTableId,
     ) -> Self::IngestBatchFuture<'_>;
 
     /// Functions the same as `ingest_batch`, except that data won't be persisted.
     fn replicate_batch(
         &self,
         kv_pairs: Vec<(Bytes, StorageValue)>,
-        epoch: u64,
+        epoch: HummockEpoch,
     ) -> Self::ReplicateBatchFuture<'_>;
 
     /// Opens and returns an iterator for given `key_range`.
     /// The returned iterator will iterate data based on a snapshot corresponding to the given
     /// `epoch`.
-    fn iter<R, B>(&self, key_range: R, epoch: u64) -> Self::IterFuture<'_, R, B>
+    fn iter<R, B>(&self, key_range: R, epoch: HummockEpoch) -> Self::IterFuture<'_, R, B>
     where
         R: RangeBounds<B> + Send,
         B: AsRef<[u8]> + Send;
 
     /// Opens and returns a reversed iterator for given `key_range`.
     /// The returned iterator will iterate data based on a snapshot corresponding to the given
     /// `epoch`
-    fn reverse_iter<R, B>(&self, key_range: R, epoch: u64) -> Self::ReverseIterFuture<'_, R, B>
+    fn reverse_iter<R, B>(
+        &self,
+        key_range: R,
+        epoch: HummockEpoch,
+    ) -> Self::ReverseIterFuture<'_, R, B>
     where
         R: RangeBounds<B> + Send,
         B: AsRef<[u8]> + Send;
 
     /// Creates a `WriteBatch` associated with this state store.
-    fn start_write_batch(&self) -> WriteBatch<Self> {
-        WriteBatch::new(self.clone())
+    ///
+    /// `table_id` is used to specify the storage group to write to.
+    fn start_write_batch(&self, table_id: StorageTableId) -> WriteBatch<Self> {
+        WriteBatch::new(self.clone(), table_id)
     }
 
     /// Waits until the epoch is committed and its data is ready to read.
-    fn wait_epoch(&self, epoch: u64) -> Self::WaitEpochFuture<'_>;
+    ///
+    /// `table_id` is used to specify the storage table to wait for when specified. If `None`, it
+    /// will wait for all tables.
+    fn wait_epoch(
+        &self,
+        table_epoch: BTreeMap<StorageTableId, HummockEpoch>,
```

Comment:
Why uses a BTreeMap here? I think wait_epoch is only called by executor and executor only needs to wait one table epoch.

Reply:
Is there any case we need to index into the b-tree map? If not, we can simply use `&[(TableId, Epoch)]` here.

Reply:
> Why uses a BTreeMap here?

Just want to provide a more flexible api in such a low layer module, in case some future code want to wait for different epochs in different table id.